src/test/java/net/fabricmc/loader/impl/discovery/GetNonFabricModsTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.impl.discovery;
+
+import static org.mockito.Mockito.mock;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedConstruction;
+import org.mockito.Mockito;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.loader.impl.FabricLoaderImpl;
+import net.fabricmc.loader.impl.game.GameProvider;
+import net.fabricmc.loader.impl.launch.FabricLauncher;
+import net.fabricmc.loader.impl.launch.FabricLauncherBase;
+
+public class GetNonFabricModsTest {
+	private FabricLoaderImpl loader;
+	private FabricLauncher launcher;
+	private ModDiscoverer discoverer;
+	private MockedConstruction<FabricLoaderImpl> loaderConstruction;
+
+	/*
+	 * Set up the mock loader and discoverer
+	 */
+	@BeforeEach
+	public void setUp() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+		GameProvider provider = mock();
+		Mockito.when(provider.getBuiltinMods()).thenReturn(Collections.emptyList());
+
+		launcher = mock();
+		Mockito.when(launcher.getEnvironmentType()).thenReturn(EnvType.CLIENT);
+		Mockito.when(launcher.isDevelopment()).thenReturn(false);
+
+		Method setLauncher = FabricLauncherBase.class.getDeclaredMethod("setLauncher", FabricLauncher.class);
```

Comment:
I dont think using reflection is tests is a good idea especially against code that can be changed, this makes the tests harder to maintain and more likely to break with future changes.

You can use the `@VisibleForTesting` annotation if needed.

You have done a really good job writing unit tests for this, considering that the original code was never desgined to be unit tested. If this was new code that I was writing it would likely be written with unit testing in mind.

Reply:
Should be fixed now.